R/arx_forecaster.R

```diff
@@ -51,13 +51,17 @@ arx_forecaster <- function(
   wf <- arx_fcast_epi_workflow(epi_data, outcome, predictors, trainer, args_list)
   wf <- fit(wf, epi_data)
 
-  preds <- forecast(
-    wf,
-    fill_locf = TRUE,
-    n_recent = args_list$nafill_buffer,
-    forecast_date = args_list$forecast_date %||% max(epi_data$time_value)
-  ) %>%
-    as_tibble() %>%
+  # get the forecast date for the forecast function
+  if (args_list$adjust_latency == "none") {
+    forecast_date_default <- max(epi_data$time_value)
+  } else {
+    forecast_date_default <- attributes(epi_data)$metadata$as_of
```

Comment:
This is about the 5th time I've seen this line. Can we make a utility:
```r
get_edf_asof <- function(x) attr(x, "metadata)$as_of
```

Reply:
I definitely think this is a good idea, but should probably be an issue over in `epiprocess`